### PR TITLE
Remove PbMapMixin, Dart mixin support

### PR DESCRIPTION
**Pull request by osa1**
This removes the last Dart mixin `PbMapMixin`. The mixin is not used
internally, and GitHub code search also doesn't find any uses other than in
forks of this repo ([1], [2]).

To keep the API surface small to make changes and migrations easier this PR
removes the mixin.

Because `PbMapMixin` was the only mixin left (after #880), this also removes
the Dart mixin support from the plugin.

[1]: https://github.com/search?q=PbMapMixin+language%3Aprotobuf&type=code
[2]: https://github.com/search?q=PbMapMixin+language%3ADart+&type=code

cl/575127174

### CODE REVIEW

File: protoc_plugin/lib/mixins.dart [hunk 1]
```diff
@@ -8,14 +8,6 @@ library;
 
 import 'indenting_writer.dart';
 
-/// Finds [name] in the exported mixins.
-PbMixin? findMixin(String name) {
```
[the remark this answers is not in the document]
Comment (osa1) [reply protoc_plugin/lib/mixins.dart:12]: `PbMixin` is still used for well-known protos: https://github.com/google/protobuf.dart/blob/master/protoc_plugin/lib/src/well_known_types.dart#L31

[the remark this answers is not in the document]
Comment (sigurdm) [reply protoc_plugin/lib/mixins.dart:12]: Ah right!

**Review by sigurdm — APPROVED** [review]

**Comment by osa1** [comment]
Note to self: clarify that this removes the built-in mixins. Users can still add arbitrary mixins to their message classes via the Dart options (https://github.com/google/protobuf.dart/blob/master/protoc_plugin/protos/dart_options.proto#L40).

**Comment by osa1** [comment]
It turns out this mixin has users. See cl/575127174 for details. Closing.

**Pull request closed without merging**